### PR TITLE
ICU-20438 Fixes issues found in the no-data testing of ICU4C (BRS hea…

### PR DESCRIPTION
**Pull request by gnrunge**
…lth check)

##### Checklist

- [ ] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20438
- [ ] Updated PR title and link in previous line to include Issue number
- [ ] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4c/source/test/intltest/dtifmtts.cpp [hunk 1]
```diff
@@ -1673,7 +1673,9 @@ void DateIntervalFormatTest::testFormattedDateInterval() {
         const char16_t* message = u"FormattedDateInterval test 1";
         const char16_t* expectedString = u"July 20 \u2013 25, 2018";
         LocalPointer<Calendar> input1(Calendar::createInstance("en-GB", status));
+	if (status.errIfFailureAndReset()) { return; }
```
Comment (markusicu) [inline icu4c/source/test/intltest/dtifmtts.cpp]: FYI ok, but only the second one is really necessary. Line 1677 won't do anything if failure, we just need to catch the failure before dereferencing the pointers on lines 1679 & 1680.

File: icu4c/source/test/intltest/dtifmtts.cpp [hunk 2]
```diff
@@ -1704,7 +1706,9 @@ void DateIntervalFormatTest::testFormattedDateInterval() {
         const char16_t* message = u"FormattedDateInterval with fallback format test 1";
         const char16_t* expectedString = u"<< July 25, 2018 --- July 20, 2018 >>";
         LocalPointer<Calendar> input1(Calendar::createInstance("en-GB", status));
+	if (status.errIfFailureAndReset()) { return; }
```
Comment (markusicu) [inline icu4c/source/test/intltest/dtifmtts.cpp]: FYI ditto

File: icu4c/source/test/intltest/measfmttest.cpp [hunk 3]
```diff
@@ -2684,7 +2684,9 @@ void MeasureFormatTest::Test20332_PersonUnits() {
     };
     for (const auto& cas : cases) {
         MeasureFormat mf(cas.locale, cas.width, status);
+	if (status.errIfFailureAndReset()) { return; }
```
Comment (markusicu) [inline icu4c/source/test/intltest/measfmttest.cpp]: FYI ditto

File: icu4c/source/test/intltest/dtifmtts.cpp (hunk 1 above)
Comment (markusicu) [inline icu4c/source/test/intltest/dtifmtts.cpp]: For several of these, the indentation is off by one. Please fix.

Comment (sffc) [reply icu4c/source/test/intltest/dtifmtts.cpp]: Seems like the indentation is tabs, but the rest of the file uses spaces.

Comment (markusicu) [reply icu4c/source/test/intltest/dtifmtts.cpp]: Tabs would be even worse...

Comment (gnrunge) [reply icu4c/source/test/intltest/dtifmtts.cpp]: Ack.

File: icu4c/source/test/intltest/dtifmtts.cpp (hunk 2 above)
Comment (gnrunge) [reply icu4c/source/test/intltest/dtifmtts.cpp]: Done.

File: icu4c/source/test/intltest/dtifmtts.cpp (hunk 1 above)
Comment (gnrunge) [reply icu4c/source/test/intltest/dtifmtts.cpp]: Done.

File: icu4c/source/test/intltest/measfmttest.cpp (hunk 3 above)
Comment (gnrunge) [reply icu4c/source/test/intltest/measfmttest.cpp]: Done.

File: icu4c/source/test/intltest/dtifmtts.cpp (hunk 1 above)
Comment (gnrunge) [reply icu4c/source/test/intltest/dtifmtts.cpp]: All indents corrected if needed.

**Pull request merged**